app/src/main/java/org/hitlabnz/sensor_fusion_demo/SensorSelectionActivity.java

```diff
@@ -1,64 +1,54 @@
-import androidx.fragment.app.FragmentManager;
-import androidx.fragment.app.FragmentPagerAdapter;
-import androidx.fragment.app.FragmentStatePagerAdapter;
-import androidx.viewpager.widget.PagerAdapter;
-import androidx.viewpager.widget.ViewPager;
+import org.hitlabnz.sensorfusionlib.orientationProvider.ImprovedOrientationSensor1Provider;
+import org.hitlabnz.sensorfusionlib.orientationProvider.OrientationProvider;
 
 /**
- * The main activity where the user can select which sensor-fusion he wants to try out
+ * The main activity
  * 
- * @author Alexander Pacha
+ * @author Ladislav Heller
  * 
  */
-public class SensorSelectionActivity extends FragmentActivity {
-
-    /**
-     * The {@link PagerAdapter} that will provide
-     * fragments for each of the sections. We use a {@link FragmentPagerAdapter} derivative,
-     * which
-     * will keep every loaded fragment in memory. If this becomes too memory
-     * intensive, it may be best to switch to a {@link FragmentStatePagerAdapter}.
-     */
-    SectionsPagerAdapter mSectionsPagerAdapter;
-
-    /**
-     * The {@link ViewPager} that will host the section contents.
-     */
-    ViewPager mViewPager;
+public class SensorSelectionActivity extends AndroidApplication {
+    private OrientationProvider op;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_sensor_selection);
 
-        // Create the adapter that will return a fragment for each of the three
-        // primary sections of the app.
-        mSectionsPagerAdapter = new SectionsPagerAdapter(getSupportFragmentManager());
+        SensorManager sm = (SensorManager)getSystemService(SENSOR_SERVICE);
+        op = new ImprovedOrientationSensor1Provider(sm);
 
-        // Set up the ViewPager with the sections adapter.
-        mViewPager = (ViewPager) findViewById(R.id.pager);
-        mViewPager.setAdapter(mSectionsPagerAdapter);
+        // create our 3D compass instance
+        // disable the use of Accelerometer/Compass/Gyroscope in LibGDX directly
+        AndroidApplicationConfiguration config = new AndroidApplicationConfiguration();
+        config.useAccelerometer = false;
+        config.useCompass = false;
+        config.useGyroscope = false;
+        config.useRotationVectorSensor = false;
+        initialize(new Compass3D(this), config);
 
         // Check if device has a hardware gyroscope
-        SensorChecker checker = new HardwareChecker((SensorManager) getSystemService(SENSOR_SERVICE));
+        // boolean gyroAvail = Gdx.input.isPeripheralAvailable(Input.Peripheral.Gyroscope);
+        SensorChecker checker = new HardwareChecker(sm);
         if(!checker.IsGyroscopeAvailable()) {
         	// If a gyroscope is unavailable, display a warning.
         	displayHardwareMissingWarning();
         }
+
+        Toast.makeText(this, "Volume Up/Down keys: Switch orientation provider!", Toast.LENGTH_LONG).show();
```

Comment:
As mentioned above: This toast disappears too quickly without being noticed and never comes back.

Reply:
> As mentioned above: This toast disappears too quickly without being noticed and never comes back.

See my above comment.